lnwallet/channel.go

```diff
@@ -291,6 +320,100 @@ type PaymentDescriptor struct {
 	isForwarded bool
 }
 
+// PayDescsFromRemoteLogUpdates converts a slice of LogUpdates received by a
+// the remote peer into PaymentDescriptors to inform a link's forwarding
+// decisions.
+func (lc *LightningChannel) PayDescsFromRemoteLogUpdates(height uint64,
+	logUpdates []channeldb.LogUpdate) []*PaymentDescriptor {
+
+	lc.Lock()
+	defer lc.Unlock()
+
+	var payDescs []*PaymentDescriptor
```

Comment:
Seems like we could pre-allocate the space to something like: 
```go
payDescsPtrs := make([]*PaymentDescriptor, 0, len(logUpdates))
payDescs := make([]PaymentDescriptor, 0, len(logUpdates))
```

Then below we'd do things like: 
```
payDescs[i] = ting
payDescPtrs[u] = &payDescs[i]
....

return payDescsPtrs
```

Reply:
Fixed